webapp/src/js/app.js

```diff
@@ -167,6 +172,12 @@ const createReduxLoggerConfig = Selectors => ({
           parsedUrl.pathname = '/dbinfo';
           url = parsedUrl.toString();
         }
+        if (opts.body && opts.body.length > BODY_LENGTH_LIMIT) {
```

Comment:
REVIEWER: I toyed with the idea of not running this check once the batch size gets to 1 so we'd attempt the upload even if we think the request is too large before failing permanently. We don't have access to the batch size param here but we could pass a request parameter to force it through... What do you think?

Reply:
Unfortunately I couldn't find a way to pass a parameter to the fetch method so there was no way to indicate that a retry was implemented.

However, once I added the server handling that you suggested it meant this code was no longer really necessary - the server responds, and the replication retries automatically. Instead I've bumped the limit to 32M characters because that's guaranteed to fail so there's no point letting it send the data to the server.

So now, if you submit less than 16M characters everything just works. If you submit more than 32M characters it fails early. If you submit between 16M and 32M characters then we try our best and fail late if necessary. I think this is a compromise solution and works well for the most common case of most of the characters being 1 byte in size.